dev/preview_docs.py

```diff
@@ -53,6 +54,24 @@ def main():
         time.sleep(3)
     else:
         print(f"Could not find {build_doc_job_name} job status")
+        comment = f"""
+### Failed to find a documentation preview link for {args.commit_sha}.
+
+<details>
+<summary>More info</summary>
+
+- If the `ci/circleci: {build_doc_job_name}` job status is successful, you can see the preview with the following steps:
+  1. Click `Details`.
+  2. Click `Artifacts`.
+  3. Click `docs/build/html/index.html`.
+- This comment was created by {workflow_run_link}.
+
+</details>
+"""
+        session.post(
+            f"https://api.github.com/repos/{repo}/issues/{args.pull_number}/comments",
+            json={"body": comment},
+        )
```

Comment:
We normally don't fall into this else branch because CircleCI triggers jobs immediately when a new commit is pushed. If we do, post a comment with a workaround.